pkg/virt-controller/services/template.go

```diff
@@ -1457,7 +1461,10 @@ func (t *templateService) VMIResourcePredicates(vmi *v1.VirtualMachineInstance,
 		vmiCPUArch = t.clusterConfig.GetClusterCPUArch()
 	}
 	memoryOverhead := GetMemoryOverhead(vmi, vmiCPUArch, t.clusterConfig.GetConfig().AdditionalGuestMemoryOverheadRatio)
-	memoryOverhead.Add(netbinding.CalculateMemoryOverhead(vmi, t.clusterConfig.GetNetworkBindings()))
+
+	if vmispec.IsVMIUsingNetworkBindingPlugins(vmi) {
+		memoryOverhead.Add(t.networkBindingPluginMemoryOverheadCalculator.CalculateMemoryOverhead(vmi))
```

Comment:
I'm commenting here because this is the main usage of the new object/method.

What is the added value in passing in an object instead of the data itself.
Now the `ClusterConfig` if embedded in a new object and previously the data itself was passed to the function directly.

Reply:
There were two reasons to have the interface to take only a `vmi`:
1. It allowed the caller to only need the `vmi` and not to have the `clusterConfig` struct.
2. It enabled this interface to be more general.

It had the cost of the `MemoryOverheadCalculator` holding a pointer to the `clusterConfig` struct, which made it a bit more complex, and required a bit more overhead in its unit test.

As discussed offline, I've changed the interface to take the registered plugins as a second argument to `CalculateMemoryOverhead` (not changing the current implementation).